src/DynamoPackages/PackageLoader.cs

```diff
@@ -397,29 +374,61 @@ public void LoadPackages(IEnumerable<Package> packages)
                 }
             }
 
-            var assemblies = packageList
+            var assemblies = packages
                 .SelectMany(p => p.LoadedAssemblies.Where(y => y.IsNodeLibrary))
                 .Select(a => a.Assembly)
                 .ToList();
+
             OnPackagesLoaded(assemblies);
         }
 
-        public void LoadCustomNodesAndPackages(LoadPackageParams loadPackageParams, CustomNodeManager customNodeManager)
+        private void LoadNewPackages(LoadPackageParams loadPackageParams)
         {
-            foreach (var path in loadPackageParams.Preferences.CustomPackageFolders)
+            foreach (var path in loadPackageParams.NewPaths)
             {
-                customNodeManager.AddUninitializedCustomNodesInPath(path, false, false);
-                if (!packagesDirectories.Contains(path))
+                var packagesDirectory = pathManager.PackagesDirectories.Where(x => x.StartsWith(path)).FirstOrDefault();
+                ScanPackageDirectories(packagesDirectory, loadPackageParams.Preferences);
+            }
+
+            if (pathManager != null)
+            {
+                foreach (var pkg in LocalPackages)
                 {
-                    packagesDirectories.Add(path);
+                    if (Directory.Exists(pkg.BinaryDirectory))
+                    {
+                        pathManager.AddResolutionPath(pkg.BinaryDirectory);
+                    }
+
                 }
             }
-            LoadAll(loadPackageParams);
+
+            if (LocalPackages.Any())
+            {
+                LoadPackages(LocalPackages);
```

Comment:
Isn't the LocalPackages array supposed to get populated by the LoadPackages call ? And now we are passing the class member LoadPackages as an argument to the sam class's method ?
or has something changed...?

Reply:
`LocalPackages` can be populated by both `ScanPackageDirectory `and `LoadPackages` call to `TryLoadPackageIntoLibrary`. If a package is already added to the list of `LocalPackages` in `ScanPackageDirectory`, it will not be added again in the subsequent call to `TryLoadPackageIntoLibrary`.

Reply:
LoadPackages is being called in several places, sometimes with a subset of packages, not necessarily with `LocalPackages` in all places. In this instance, we need to call it with the same class member but that is not always the case.

Reply:
Ok, I get it now. Thanks